src/controllers/hid/hidcontroller.cpp

```diff
@@ -194,22 +194,55 @@ QByteArray HidController::getInputReport(unsigned int reportID) {
-    // the mapping JS code takes longer than the controller polling rate.
-    // This could stall other low priority tasks.
-    // There is no safety net for this because it has not been demonstrated to be
-    // a problem in practice.
+    // This function reads the available HID Input Reports using hidapi.
+    // Important to know is, that this reading is not a hardware operation,
+    // instead it reads previously received HID Input Reports from a ring buffer.
+    // Depending on the hidapi backend implementation the ring buffer is either part
+    // of the hidapi implementation or the OS kernel.
+    // The size of the ring buffer also depends on the hidapi backend:
+    // - hidraw(2048 bytes) - Linux Kernel API
+    // - libusb(30 reports) - BSD (alternative Linux userspace implementation)
+    // - mac(30 reports)
+    // - windows(64 reports)
+    // Behavior:
+    // - If the polling rate is to slow, multiple HID Input Reports will be processed at the same time
+    // - If the mapping processes the Input Reports too slow, whole Input Reports will be omitted if the ring buffer is full
+    // - If the processing of all received HID Input Reports in the JS mapping code takes longer than the interval between incoming reports,
+    //   it could stall other low priority tasks.
+
+    constexpr int kSmallestRingBufferInReports = 30;
+    constexpr int kSmallestRingBufferInBytes = 2048;
+
+    int inputReportCounter = 0;
+    int inputReportByteCounter = 0;
     while (true) {
         int bytesRead = hid_read(m_pHidDevice, m_pPollData[m_pollingBufferIndex], kBufferSize);
         if (bytesRead < 0) {
             // -1 is the only error value according to hidapi documentation.
+            qCWarning(m_logOutput) << "Unable to read HID Input Reports from" << getName() << ":"
+                                   << mixxx::convertWCStringToQString(
+                                              hid_error(m_pHidDevice),
+                                              kMaxHidErrorMessageSize);
             DEBUG_ASSERT(bytesRead == -1);
             return false;
         } else if (bytesRead == 0) {
-            // No packet was available to be read
+            // No packets left to be read
+
+            // If we are below both limits, we can say safely that no packages are lost.
+            // If we are at or above the limits, it can be, that reports are dropped. -> Print warning
+            if ((inputReportCounter >= kSmallestRingBufferInReports) ||
+                    (inputReportByteCounter >= kSmallestRingBufferInBytes)) {
+                qCWarning(m_logOutput)
+                        << getName() << ": Read "
+                        << inputReportCounter
+                        << " HID Input Reports with a combined size of " << inputReportByteCounter
+                        << "in the same polling interval. This indicates a "
+                           "performance problem with controller mapping.";
```

Comment:
It's also possible that the computer is just old and slow, right? Or is this very unlikely?